app/src/main/java/com/RaphaelAGN/chucknorrisapp/MainActivity.kt

```diff
@@ -2,53 +2,56 @@ package com.RaphaelAGN.chucknorrisapp
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.widget.Button
-import android.widget.TextView
-import android.widget.Toast
-import com.RaphaelAGN.chucknorrisapp.endpoint.Endpoint
-import com.RaphaelAGN.chucknorrisapp.data.JokeModel
-import com.RaphaelAGN.chucknorrisapp.retrofitClient.RetrofitClient
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.Response
+import android.util.Log
+import androidx.fragment.app.*
+import com.RaphaelAGN.chucknorrisapp.fragments.MainFragment
 
 class MainActivity : AppCompatActivity() {
 
-    var jokeButton: Button? = null
-    var jokeTextView: TextView? = null
+    val manager: FragmentManager = supportFragmentManager
+    val transaction: FragmentTransaction = manager.beginTransaction()
```

Comment:
Avoid to call `beginTransaction()` before it needed. It's better to call it right before the `replace`